content/python/pagination.py.md

```diff
@@ -1,52 +1,143 @@
+
+The python client itself has a built in function called [`iter_call`](https://softlayer-python.readthedocs.io/en/latest/api/client/#SoftLayer.API.BaseClient.iter_call) which will return a python [generator](https://wiki.python.org/moin/Generators). This generator will only get the next set of results when you need them, which can make doing operations on large datasets feel faster in that you don't have to wait for the entire dataset to be retrieved before obtaining some results.
 
-```
-"""
-Example about pagination
 
-The script retrieves the Bare metal servers in a account
-and displays all the servers using pagination
+## [The Setup](#the-setup) {#the-setup .anchor-link}
 
-Important manual pages
-http://sldn.softlayer.com/reference/services/SoftLayer_Account
-http://sldn.softlayer.com/reference/services/SoftLayer_Account/getHardware
-https://softlayer-api-python-client.readthedocs.org/en/latest/api/client/#module-SoftLayer
+To avoid having to type the boiler-plate code a lot, assume each of these example function will be a member function of the following `Paginator` class
 
-License: http://sldn.softlayer.com/article/License
-Author: SoftLayer Technologies, Inc. <sldn@softlayer.com>
-"""
+```python
 import SoftLayer
 
-# Your SoftLayer API username and key.
-USERNAME = 'set me'
-API_KEY = 'set me'
-
-client = SoftLayer.Client(username=USERNAME, api_key=API_KEY)
-
-offset = 0
-limit = 1
-
-accountService = client['SoftLayer_Account']
-
-while True:
-    try:
-        result = accountService.getHardware(limit=limit, offset=offset)
-        offset = offset + limit
-        limit = limit + limit
-        print(result)
-        if not result:
-            break
-    except SoftLayer.SoftLayerAPIError as e:
-        print("Unable to retrieve the servers . " % (e.faultCode, e.faultString))
-    exit(1)
-    
+class Paginator():
+
+    def __init__(self):
+        self.client = SoftLayer.create_client_from_env()
+        debugger = SoftLayer.DebugTransport(self.client.transport)
+        self.client.transport = debugger
+        self.DESC = {
+            "operation": "orderBy", 
+            "options": [{
+                "name": "sort", 
+                "value": ["DESC"]
+            }]
+        }
+
+if __name__ == "__main__":
+    paginator = Paginator()
+    # Then just call whichever function you want to run like this
+    # paginator.list_servers()
 ```
+
+## [A simple pagination example](#simple-example) {#simple-example .anchor-link}
+
+```python
+
+def list_servers(self):
+    orderBy = {"hardware": {"id": self.DESC}}
+    objectMask = "mask[id,hostname]"
+    servers = self.client.iter_call('SoftLayer_Account', 'getHardware', filter=orderBy, mask=objectMask, limit=10)
```

Comment:
only 10 servers, is possible in the method parameter send this servers number to getting

Reply:
yes, but I'll leave that as an exercise for the reader.